apps/rush-lib/src/logic/taskRunner/ProjectBuilder.ts

```diff
@@ -210,16 +210,25 @@ export class ProjectBuilder extends BaseBuilder {
 
       let projectBuildCache: ProjectBuildCache | undefined;
       if (this._buildCacheConfiguration) {
-        const projectBuildCacheConfiguration: ProjectBuildCacheConfiguration = ProjectBuildCacheConfiguration.loadForProject(
+        const projectBuildCacheConfiguration:
+          | ProjectBuildCacheConfiguration
+          | undefined = await ProjectBuildCacheConfiguration.tryLoadForProjectAsync(
           this._rushProject,
-          this._buildCacheConfiguration
+          terminal
         );
-        projectBuildCache = this._buildCacheConfiguration.cacheProvider.tryGetProjectBuildCache(terminal, {
-          projectBuildCacheConfiguration: projectBuildCacheConfiguration,
-          command: this._commandToRun,
-          projectBuildDeps: projectBuildDeps,
-          packageChangeAnalyzer: this._packageChangeAnalyzer
-        });
+        if (projectBuildCacheConfiguration) {
+          projectBuildCache = this._buildCacheConfiguration.cacheProvider.tryGetProjectBuildCache(terminal, {
+            projectBuildCacheConfiguration: projectBuildCacheConfiguration,
+            command: this._commandToRun,
+            projectBuildDeps: projectBuildDeps,
+            packageChangeAnalyzer: this._packageChangeAnalyzer
+          });
+        } else {
+          terminal.writeVerboseLine(
+            'Project does not have a build-cache.json configuration file, or one provided by a rig, ' +
+              'so it does not support caching.'
```

Comment:
What happens in this case?  Seems like it should prevent downstream (dependant) projects from getting cached as well.

Reply:
You think so? I'm struggling to come up with a case where a project should care about its upstreams' cache configurations.

It might be a little weird to have to rebuild a project every time and get its downstream projects from cache, but it doesn't seem incorrect.

Reply:
@iclanton explained that the downstream project's cache key is calculated using the upstream dependency's Git-tracked assets and dependencies (not its build outputs).  Thus it does seem valid to cache a downstream project.